File: cfgs/java_cfgs_foo/CHECK_POSSIBLE_SORT_ARRAY_CONDITIONAL_SWAPPING_ADJACENT_ALLOWED.java
Nodes
1;special;;;
2;statement;int i = 0;14;
3;conditional;foo_1 < (foo_2 - 1);15;
4;conditional;foo_1[foo_2] > foo_3[foo_4 + 1];17;
5;conditional;(foo_1[foo_2] - foo_3[foo_4 + 1]) == 1;18;
6;statement;int temp = foo_1[foo_2];19;
7;statement;foo_1[foo_2] = foo_3[foo_4 + 1];20;
8;statement;foo_1[foo_2 + 1] = foo_3;21;
9;statement;return false;23;
10;exit;;;
11;statement;i++;16;
12;statement;return true;26;
Edges
0;2;
1;3;
2;4;12;
3;5;11;
4;6;9;
5;7;
6;8;
7;11;
8;10;
9;
10;3;
11;10;

File: cfgs/pythoncfgs/CHECK_POSSIBLE_SORT_ARRAY_CONDITIONAL_SWAPPING_ADJACENT_ALLOWED.py
Nodes
1;conditional;foo < foo;8;
2;statement;return true;14;
3;conditional;foo_1[foo_2] > foo_3[foo_4 + 1];9;
4;conditional;foo_1[foo_2] - foo_3[foo_4 + 1] == 1;10;
5;statement;return false;13;
6;statement;(foo_1[foo_2], foo_3[foo_4 + 1]) = (foo_5[foo_6 + 1], foo_7[foo_8]);11;
7;exit;;
Edges
0;2;3;
1;7;
2;1;4;
3;5;6;
4;7;
5;1;
6;